src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -119,6 +119,48 @@ AppendAlterTableCmd(StringInfo buf, AlterTableCmd *alterTableCmd)
 			break;
 		}
 
+		case AT_AddConstraint:
+		{
+			Constraint *constraint = (Constraint *) alterTableCmd->def;
+
+			/* We need to deparse ALTER TABLE ... PRIMARY KEY commands into
+			 * ALTER TABLE ... ADD CONSTRAINT <conname> PRIMARY KEY ... to be able
+			 * add a constraint name.
+			 * TODO: this code needs to support all possible syntaxes for ALTER TABLE ... PRIMARY KEY
+			 * (e.g. UNCLUSTERED)
+			 */
+			if (constraint->contype == CONSTR_PRIMARY)
+			{
+				/* Need to deparse PRIMARY KEY constraint commands only if adding a name.*/
+				Assert(constraint->conname != NULL);
```

Comment:
So, we don't want to crash here but rather just error out with
```
ereport(ERROR, (errmsg("unsupported subtype for alter table command")
```

Reply:
Hmm makes sense, but maybe we can error out earlier? This deparswr is very limited and in other parts of the deparser we have just thrown an error (the error is also not supposed to ever be reached by any user scenarios). I was thinking we can be consistent with the other parts and throw the error.

Otherwise we can put Asserts in the other parts of the deparser as well, but I guess it's not the practice we usually follow (I checked the other deparsers a bit as well)

Reply:
> but maybe we can error out earlier?

We cannot error out earlier so some place here would be the appropriate place to error out